energy-integration/common-modules/weekly-rewards-splitting/src/lib.rs

```diff
@@ -72,7 +74,7 @@ pub trait WeeklyRewardsSplittingModule:
             let weeks_to_claim = core::cmp::min(total_weeks_to_claim, MAX_CLAIM_PER_TX);
             for _ in 0..weeks_to_claim {
                 let rewards_for_week =
-                    self.claim_single(user, current_week, collect_rewards_fn, &mut claim_progress);
+                    self.claim_single::<WRSM>(wrapper, user, current_week, &mut claim_progress);
```

Comment:
I don't think you need to specify the wrapper type here, the compiler should deduce it from `wrapper`'s type.
```suggestion
                    self.claim_single(wrapper, user, current_week, &mut claim_progress);
```

Reply:
Done.